drivers/hmis/app/models/hmis/hud/assessment.rb

```diff
@@ -75,14 +77,16 @@ def save_in_progress
 
     self.enrollment_id = WIP_ID
     save!(validate: false)
-    self.wip = Hmis::Wip.find_or_create_by(
+    self.wip = Hmis::Wip.create_with(date: assessment_date).find_or_create_by(
```

Comment:
This is awesome.  I wasn't familiar with `create_with`
https://apidock.com/rails/v6.1.3.1/ActiveRecord/QueryMethods/create_with
I wonder how this interacts with the block syntax for `find_or_create`
I've always done `Hmis::Wip.where(source: self).find_or_create do |wip|
  wip.date = assessment_date
end
```
but I think this is way more clear.